docs/installation.rst

```diff
@@ -0,0 +1,56 @@
+Installation & setup
+======================
+
+
+You can install the module form PyPI (or from git by the standard commands).
+
+.. code-block:: bash
+
+    python -m pip install python_x509_pkcs11
+
+
+Setup of OS packages and environment variables
+-----------------------------------------------
+
+We will need the module for the HSM device and the secret PIN code and the
+`Token Name` in the HSM device. For the development we will use `softhsm
+<https://www.opendnssec.org/softhsm/>`_ to emulate a physical device.
+
+The following script will help you to install the packages and setup the environment.
+
+.. code-block:: bash
+
+    if awk -F= '/^NAME/{print $2}' /etc/os-release | grep -i "debian\|ubuntu"
+    then
+        # Ubuntu / Debian
+        sudo apt-get install python3-dev python3-pip softhsm2
+        sudo usermod -a -G softhsm $USER
+    else
+        # Redhat / Centos / Fedora
+        sudo dnf install python3-devel python3-pip softhsm gcc 
+        sudo usermod -a -G ods $USER
+    fi
+
+    # Update your softhsm group membership
+    exec sudo su -l $USER
+
+    # export env values the code will use
+    if awk -F= '/^NAME/{print $2}' /etc/os-release | grep -i "debian\|ubuntu"
```

Comment:
Repeated awk on line 23, maybe declare variable OS=$(awk -F...) and compare it?

Reply:
Does this script exists somewhere?

Reply:
This is direct copy paste from the existing documentation, I say we update all of these instructions in a future release.